pkg/kinflate/configmapandsecret/configmap_secret.go

```diff
@@ -224,18 +170,13 @@ func MakeGenericSecretsKObject(secrets []manifest.GenericSecret) (types.KObject,
 	return m, nil
 }
 
-// MakeTLSSecretsKObject returns a map of <GVK, oldName> -> unstructured object.
-func MakeTLSSecretsKObject(secrets []manifest.TLSSecret) (types.KObject, error) {
-	m := types.KObject{}
-	for _, secret := range secrets {
-		unstructuredSecret, nameWithHash, err := MakeTLSSecretAndGenerateName(secret)
-		if err != nil {
-			return nil, err
-		}
-		err = populateMap(m, unstructuredSecret, nameWithHash)
-		if err != nil {
-			return nil, err
-		}
+func createSecretKey(wd string, command string) ([]byte, error) {
+	fi, err := os.Stat(wd)
+	if err != nil || !fi.IsDir() {
+		wd = filepath.Dir(wd)
 	}
-	return m, nil
+	cmd := exec.Command("sh", "-c", command)
+	cmd.Dir = wd
```

Comment:
suppose my secret generator command is "sleep 5000"
or the command could try to run a program that isn't installed, etc.

need timeout and error handling, e.g. start a go routine the timeouts after a while, and this code selects on it or command completion.  The test could literally be a sleep longer than whatever you select as a timeout.

want to do in this PR or file an issue to add it later?

Reply:
CommandContext fits this use case pretty well.
https://golang.org/pkg/os/exec/#CommandContext

Reply:
I'll update now.